plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplit.java

```diff
@@ -136,6 +140,13 @@ public List<TrinoDeleteFile> getDeletes()
         return deletes;
     }
 
+    @JsonProperty
+    @Override
+    public SplitWeight getSplitWeight()
+    {
+        return splitWeight;
```

Comment:
`SplitWeight.fromProportion(length)`

Reply:
besides being simpler, it's better to avoid a new field, so that it's clear what the weight is, and that it's reasonable & consistent with other split's state

Reply:
We could try `SplitWeight.fromRawValue(length)` but that comes with the implication that `100` is the value of a "standard size split" which isn't right for length.  I think if we do that the weights will imply that regular size splits are very large and may be scheduled slower. I haven't tested it though.

Reply:
My understanding from skimming the code is that if all splits return `fromRawValue(100)`, or `fromProportion(1.0)` splits will be scheduled exactly how they are now, which is what we want for full size splits.

Reply:
To address the initial feedback point, you need to know the split's length in bytes relative to the configured target split size, which means that you would need to compute the value and store it as a field unless the target split size was also hard-coded.

Reply:
I now understand this isn't a good idea.

> To address the initial feedback point, you need to know the split's length in bytes relative to the configured target split size

that's weird concept from SPI perspective, since "configured target split size" is not a SPI concept itself


@pettyjamesm if a connector uses `fromProportion(d)` is relation between `d` and `1.0` really important?
how?

(and where is it documented? :) )

Reply:
A split with a weight of `1.0` is "standard". If all splits are standard weight, then the scheduler will place exactly as many splits per node per task as the scheduler config specifies. If all splits have a weight of `0.5`, then the scheduler is allowed to place 2x as many splits per node and task, and with all splits weighing `2.0`- 1/2 as many.

It's not explicitly documented, although the PrestoDB PR which merged after the Trino PR did include references to this behavior in the [NodeScheduler Properties Reference](https://prestodb.io/docs/current/admin/properties.html#node-scheduler-max-splits-per-node). We should probably port those documentation changes.

Reply:
> We should probably port those documentation changes.

thanks, please do